test/client-test.js

```diff
@@ -1081,7 +1081,7 @@ var fs = require('fs'),
       var baseUrl = 'http://' + hostname + ':' + port;
 
       server = http.createServer(function (req, res) {
-        res.statusCode = 200;
```

Comment:
i don't think this should be changed to 500.  please revert

Reply:
W3C spec says SOAP faults should return HTTP status 500, because this test case doesn't actually test for the spec this is why this PR exists because the SOAP faults are being bubbled up since 0.40.0.  Every SOAP service I have worked with returns SOAP faults with HTTP status 500.  https://www.w3.org/TR/2000/NOTE-SOAP-20000508/#_Toc478383529

Reverting this will not fix the failing test cases.

Reply:
Split the difference, let me know what you thing about the latest revision.  Added a loop to test soap faults are working for both status 200 & 500.